psw/urts/tcs.cpp

```diff
@@ -150,11 +150,15 @@ int CTrustThreadPool::bind_thread(const se_thread_id_t thread_id,  CTrustThread
 
 CTrustThread * CTrustThreadPool::get_bound_thread(const se_thread_id_t thread_id)
 {
-    CTrustThread *trust_thread = NULL;
+    CTrustThread *trust_thread = nullptr;
 
-    Node<se_thread_id_t, CTrustThread*>* it = m_thread_list->Find(thread_id);
-    if(it != NULL)
-        trust_thread = it->value;
+    if(m_thread_list)
```

Comment:
You need to check for m_thread_list == nullptr here. Everything else is not standard behavior. GCC6 optimized this away and the simulator crashes... Took me a while to figure this out...